apps/web/src/middlewares/ab-test-middleware.ts

```diff
@@ -0,0 +1,67 @@
+import { AB_TESTING_FEATURE_FLAG_MAP } from 'contexts/ABTestingContext/config'
+import { NextFetchEvent, NextRequest, NextResponse } from 'next/server'
+import { FeatureFlagInfo, MiddlewareFactory } from './types'
+
+// this function generates a deterministic result for a user for a given feature
+// it hashes a concatination of the users ip together with the features identifier and
+// probanility value. this allows us to ensure that a users probability result is different
+// for each feature gauranteeing a better distribution
+export const generateUserDeterministicValue = async (
```

Comment:
Naming suggestion. 

The function name should always explain the purpose of this function instead of describing how it's done. When I see `generateUserDeterministicValue`, questions pop into my mind, like what is a deterministic value and why are we trying to generate that. You write total 4 lines of comment to explain that and these are all detailed implementation that the caller doesn't need to know. 

For the caller, he doesn't care about how you implement this function. All he need is to get the list of the ongoing experimental features and check if the user has access to them

```suggestion
export const getExperimentalFeatureAccessList = async (
```

Reply:
I'm not saying that we don't need these comments, but we should put them inside the function for further explanation, as the caller doesn't care